### DISCUSSION TITLE
Print to stdout

### DISCUSSION THREAD

**Discussion by dskarbrevik** [Textualize/textual#3085] [General]
Beginner question: how can I get print messages to display to my terminal while my Textual app is running?

It seems they're eaten by Textual and in my environment I cannot run  the textual console server. So how can I ensure that I see print statements in this case?

**Comment by willmcgugan** [comment]
> It seems they're eaten by Textual and in my environment I cannot run the textual console server

I assume you are referring to `textual console`? Why can't you run it?

**Reply by dskarbrevik** [reply]
Thanks for the reply Will.

My environment uses the server port that the textual console wants to use and I don't have control over it. Further I don't need to debug anything about my textual app. I need to debug cloud logging utilities that I happen to be putting in the app code.

So for my scenario it would be fastest to iterate by seeing a log or print statement in terminal.

I've also tried `print("something", flush=True)` which didn't work for me.

**Reply by willmcgugan** [reply]
You can change the port with the `--port` switch on both the console and `textual run`:

`textual console --port 8001`

`textual run --port 8001 myapp.py` 

Or you can set `TEXTUAL_PORT=8001` in your environment.

But if a plain text log suits you , you can set the env var `TEXTUAL_LOG=textual.log`
